x/auctions/keeper/hooks.go

```diff
@@ -17,50 +18,53 @@ func (k Keeper) BeforeEpochStart(_ sdk.Context, _ string, _ int64) {
 
 // AfterEpochEnd ends the current auction and distributes the rewards to the winner
 func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, _ int64) {
+	// Create a cached context that is committed only
+	// if not errors happen.
+	ctxCache, writeFn := ctx.CacheContext()
```

Comment:
move this inside the `if`block. Outside it we're not using it

Reply:
Nice catch!